phaser/steps.py

```diff
@@ -44,6 +48,26 @@ def _batch_step_wrapper(batch, context=None, __probe__=None):
     return _batch_step_wrapper
 
 
+def dataframe_step(step_function):
+    @wraps(step_function)
+    def _df_step_wrapper(row_data, context=None, __probe__=None):
+        if __probe__ == PROBE_VALUE:
+            return DATAFRAME_STEP
+        try:
+            dataframe = pd.DataFrame.from_records(row_data)
```

Comment:
Brilliant! Isolating the conversion to and from DataFrame in the wrapper will make removing the Pandas dependency much easier.